config/config.go

```diff
@@ -250,13 +250,52 @@ func (c *Config) Validate() error {
 		return err
 	}
 
+	if err := c.validateTaskProvider(); err != nil {
+		return err
+	}
+
 	if err := c.validateDynamicConfigs(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// validateTaskProvider checks that task <-> provider relations are good
+func (c *Config) validateTaskProvider() error {
+	// provider name -> auto-commit setting, the 2 things we need to check
+	providers := make(map[string]bool, len(*c.TerraformProviders))
+	for _, p := range *c.TerraformProviders {
+		for name, values := range *p {
+			if a, ok := values.(map[string]interface{})["alias"]; ok {
+				name = fmt.Sprintf("%s.%s", name, a)
+			}
+			providers[name] = false
+			if v, ok := values.(map[string]interface{})["auto-commit"]; ok {
```

Comment:
Most of the Terraform provider arguments I've seen are underscore delimited. Since this would be a part of the `terraform_provider` blocks could you change it to `auto_commit` for consistency?

Reply:
One caution to consider when adding a CTS-specific option to the `terraform_provider` block is the potential of overlap with an existing Terraform provider option.

For example `task_env` does have a small chance of overlap, but I don't anticipate it happening. Auto commit seems like a more common term in general which increases overlap.

Reply:
I wasn't sure between auto-commit, auto_commit and autocommit... and I couldn't seem to find an example of another 2-word name so I just guessed knowing it'd come up in the review if neededd. :)

Reply:
auto_commit has been committed